include/nanobind/nanobind.h

```diff
@@ -32,6 +32,8 @@
 #include <typeinfo>
 #include <utility>
 #include <new>
+#include <optional>
```

Comment:
There is no way that we can include these heavy C++ headers in `nanobind.h` itself. See this excerpt from README.md:

> `#include <pybind11/pybind11.h>` pulls in a large portion of the STL (about 2.1 MiB of headers with Clang and libc++). nanobind minimizes STL usage to avoid this problem. Type casters even for for basic types like std::string require an explicit opt-in by including an extra header file (e.g. `#include <nanobind/stl/string.h>`).